packages/toolpad-app/src/toolpad/AppEditor/StructureExplorer/index.tsx

```diff
@@ -187,40 +213,74 @@ export default function PageStructureExplorer() {
-      });
+      setExpandedDomNodeIds(nodeIds);
     },
     [setExpandedDomNodeIds],
   );
 
+  const handleNodeDelete = React.useCallback(
+    (event: React.KeyboardEvent<HTMLUListElement>) => {
+      // delete selected node if event.key is Backspace
+      if (event.key === 'Backspace') {
+        appStateApi.update(
+          (draft) => {
+            if (!selectedDomNodeId) {
+              return draft;
+            }
+            const toRemove = appDom.getMaybeNode(dom, selectedDomNodeId);
+            if (toRemove && appDom.isElement(toRemove)) {
+              draft = appDom.removeNode(draft, toRemove.id);
+              draft = deleteOrphanedLayoutNodes(dom, draft, toRemove);
+            }
+            return draft;
+            // return normalizePageRowColumnSizes(draft);
+          },
+          currentView.kind === 'page'
+            ? {
+                ...currentView,
+                selectedNodeId: null,
+              }
+            : currentView,
+        );
+      }
+    },
+    [dom, selectedDomNodeId, currentView, appStateApi],
+  );
+
   const expandedDomNodeIdSet = React.useMemo(() => {
-    return new Set([selectedNodeParentId as string, ...expandedDomNodeIds]);
-  }, [selectedNodeParentId, expandedDomNodeIds]);
+    return new Set([...selectedNodeAncestorIds, ...expandedDomNodeIds]);
+  }, [selectedNodeAncestorIds, expandedDomNodeIds]);
 
   return (
-    <TreeView
-      aria-label="file system navigator"
-      defaultCollapseIcon={<ExpandMoreIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
-      defaultExpandIcon={<ChevronRightIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
-      expanded={Array.from(expandedDomNodeIdSet)}
-      selected={selectedDomNodeId as string}
-      onNodeSelect={handleNodeSelect}
-      onNodeFocus={handleNodeFocus}
-      onNodeToggle={handleNodeToggle}
-      sx={{ height: 600, flexGrow: 1, maxWidth: 400, overflowY: 'auto' }}
-    >
-      {rootChildren.map((childNode) => (
-        <RecursiveSubTree
-          key={childNode.id}
-          dom={dom}
-          root={childNode}
-          onHover={handleNodeHover}
-          onMouseLeave={handleNodeBlur}
-        />
-      ))}
-    </TreeView>
+    <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'start', p: 1 }}>
+      <Typography
+        variant="body2"
+        sx={(theme) => ({ flexGrow: 1, fontWeight: theme.typography.fontWeightLight, pb: 0.5 })}
+      >
+        Components
+      </Typography>
+      <TreeView
+        aria-label="components explorer"
+        defaultCollapseIcon={<ExpandMoreIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
+        defaultExpandIcon={<ChevronRightIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
+        expanded={Array.from(expandedDomNodeIdSet)}
+        selected={selectedDomNodeId as string}
+        onNodeSelect={handleNodeSelect}
+        onNodeFocus={handleNodeFocus}
+        onNodeToggle={handleNodeToggle}
+        onKeyDown={handleNodeDelete}
```

Comment:
I would separate `onKeyDown` to something like `handleKeyDown`, and there is backspace is pressed call a reusable delete function. 
That way for example if we allow deleting in the explorer by clicking the button we just have to use the same function that is already there.